CUDADataFormats/Track/interface/TrackSoAHeterogeneousHost.h

```diff
@@ -0,0 +1,32 @@
+#ifndef CUDADataFormats_Track_TrackHeterogeneousHost_H
+#define CUDADataFormats_Track_TrackHeterogeneousHost_H
+
+#include <cstdint>
+
+#include "CUDADataFormats/Track/interface/PixelTrackUtilities.h"
+#include "CUDADataFormats/Common/interface/PortableHostCollection.h"
+
+template <typename TrackerTraits>
+class TrackSoAHeterogeneousHost : public cms::cuda::PortableHostCollection<TrackLayout<TrackerTraits>> {
```

Comment:
For the next round (CUDA-to-Alpaka?) I'm wondering if the "Host" postfix could be dropped from the host-side data format class names.

My rationale is that these are the data formats seen by the downstream non-CUDA/Alpaka consumers, and for them the `Host` postfix feels somewhat unnecessary. Or would `Foo` and `FooDevice` be too confusing? (also keeping in mind that currently the `PortableCollection` infrastructure has the `Host` explicitly in `PortableHostCollection` name)

Reply:
From my (completely new to the whole framework, who's working only temporarily with it) perspective, having `Host` in the type name makes everything _much_ more clear and straightforward. It states that:
- It's on `Host`, hence you can make sure at any time that the data are where you expect them to be,
- It indirectly states that there is also a `Device` one, meaning that the data is a `PortableCollection`.

Reply:
I think the alpaka case is going to be slightly more complicated.

We could have
  - `ALPAKA_ACCELERATOR_NAMESPACE::TrackSoA` (or `TrackSoAHost`) for the host collection
  - `ALPAKA_ACCELERATOR_NAMESPACE::TrackSoADevice` for the device collection

_but_ `TrackSoA`(`Host`) and `TrackSoADevice` need to be the same type when the "device" is actually the CPU.

This was easy to do in `DataFormats/PortableTestObjects/interface/` because we use the types directly, without inheritance or composition.
We will need to understand how to do it for these more complex cases.